src/main/kotlin/uk/gov/justice/hmpps/offenderevents/services/HMPPSDomainEventsEmitter.kt

```diff
@@ -505,6 +507,46 @@ class HMPPSDomainEventsEmitter(
 
       else -> emptyList()
     }
+
+  private fun VideoAppointmentCancelledEvent.Companion.toDomainEvents(event: VideoAppointmentCancelledEvent): List<HmppsDomainEvent> =
+    event.toDomainEvent().toListOrEmptyWhenNull()
+
+  private val videoAppointmentTypes = listOf("VLB", "VLPM", "VLLA", "VLOO", "VLPA", "VLAP")
+
+  private fun VideoAppointmentCancelledEvent.toDomainEvent(): HmppsDomainEvent? =
+    when {
+      // Only raise an event for the 6 video appointment types, and only if they have been cancelled or deleted
+      (this.recordDeleted || this.scheduleEventStatus == "CANC") && videoAppointmentTypes.contains(this.scheduleEventSubType) -> {
+        prisonApiService.getPrisonerNumberForBookingId(this.bookingId).getOrNull()?.let {
+          HmppsDomainEvent(
+            eventType = "prison-offender-events.video-appointment.cancelled",
+            description = "A video appointment has been cancelled",
+            occurredAt = this.toOccurredAt(),
+            publishedAt = OffsetDateTime.now().toString(),
+            personReference = PersonReference(it),
+          )
+            .withAdditionalInformation("bookingId", this.bookingId)
+            .withAdditionalInformation("scheduleEventId", this.scheduleEventId)
+            .withAdditionalInformation("scheduledStartTime", this.scheduledStartTime.toString())
+            .withAdditionalInformation("scheduledEndTime", this.scheduledEndTime.toString())
+            .withAdditionalInformation("scheduleEventType", this.scheduleEventType)
+            .withAdditionalInformation("scheduleEventSubType", this.scheduleEventSubType)
+            .withAdditionalInformation("scheduleEventStatus", this.scheduleEventStatus)
+            .withAdditionalInformation("recordDeleted", this.recordDeleted)
+            .withAdditionalInformation("agencyLocationId", this.agencyLocationId)
+        }
+      }
+      else -> {
+        log.debug(
```

Comment:
I'd prefer not to log every single appointment changed event here too - will already be logged in domain events as it is

Reply:
Ok - happy to remove - tests cover the case where it shouldn't raise events, so I agree its not needed.

Reply:
Removed this logging.